crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1481,31 +1538,41 @@ impl AuthorityPerEpochStore {
             iter::once((tx_digest, assigned_versions)),
         )?;
 
-        self.finish_consensus_certificate_process_with_batch(
-            write_batch,
-            key,
-            certificate,
-            consensus_index,
-        )?;
+        self.finish_consensus_certificate_process_with_batch(write_batch, certificate)?;
         Ok(())
     }
 
-    /// When we finish processing certificate from consensus we record this information.
-    /// Tables updated:
-    ///  * consensus_message_processed - indicate that this certificate was processed by consensus
-    ///  * last_consensus_index - records last processed position in consensus stream
-    /// Self::consensus_message_processed returns true after this call for given certificate
-    fn finish_consensus_transaction_process_with_batch(
+    /// Record when finished processing a transaction from consensus.
+    fn record_consensus_message_processed(
         &self,
         batch: &mut DBBatch,
         key: SequencedConsensusTransactionKey,
-        consensus_index: &ExecutionIndicesWithHash,
     ) -> SuiResult {
+        batch.insert_batch(&self.tables.consensus_message_processed, [(key, true)])?;
+        Ok(())
+    }
+
+    /// Record when finished processing a consensus commit.
+    fn record_consensus_commit_stats(
+        &self,
+        batch: &mut DBBatch,
+        consensus_stats: &ExecutionIndicesWithStats,
+    ) -> SuiResult {
+        // TODO: remove writing to last_consensus_index.
         batch.insert_batch(
             &self.tables.last_consensus_index,
-            [(LAST_CONSENSUS_INDEX_ADDR, consensus_index)],
+            [(
+                LAST_CONSENSUS_STATS_ADDR,
+                ExecutionIndicesWithHash {
+                    index: consensus_stats.index,
+                    hash: consensus_stats.hash,
+                },
+            )],
+        )?;
```

Comment:
Maybe it is unnecessary, but I'm trying to support rolling back binary as well, because this change is not guarded by protocol version. The cost should be minimal.